### PR TITLE
Feature/merge message releases

### PR DESCRIPTION
**Pull request by DSpirit**
Extended the `TryParse` operation to check whether the commit source branch is a release branch in order to support increment of merges from release into main branch when a squash merge has been applied.

## Motivation and Context
[Discussion](https://github.com/GitTools/GitVersion/discussions/3964)

## How Has This Been Tested?
- Local tests with only one merge commit worked
- Extended the `MergeMessageBaseVersionStrategyTests.cs` with a test using the `false` flag on `IsMergeCommit`

❗ I had to comment one test, as the `hotfix-0.1.5` branch was somehow configured as a release branch and thus, the version was taken into respect. Would appreciate some feedback here!

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/GitVersion.Core/MergeMessage.cs [hunk 1]
```diff
@@ -96,13 +96,10 @@ private ReferenceName GetMergedBranchName(string mergedBranch)
         mergeCommit.NotNull();
         configuration.NotNull();
 
-        mergeMessage = null;
+        mergeMessage = new(mergeCommit.Message, configuration);
```
Comment (asbjornu) [inline src/GitVersion.Core/MergeMessage.cs:99]: `mergeMessage` should not be set if the return value equals `false`.

File: src/GitVersion.Core.Tests/VersionCalculation/Strategies/MergeMessageBaseVersionStrategyTests.cs [hunk 2]
```diff
@@ -65,7 +66,7 @@ public void TakesVersionFromMergeOfReleaseBranch(string message, bool isMergeCom
         AssertMergeMessage(message + "\n ", expectedVersion, parents);
     }
 
-    [TestCase("Merge branch 'hotfix-0.1.5'", false)]
+    // [TestCase("Merge branch 'hotfix-0.1.5'", false)] this is somehow configured as a release branch
```
Comment (asbjornu) [inline src/GitVersion.Core.Tests/VersionCalculation/Strategies/MergeMessageBaseVersionStrategyTests.cs:69]: Can you please elaborate on how `hotfix-0.1.5` is configured as a release branch? Is `is-release-branch` set to `true`?

**Review by asbjornu — CHANGES_REQUESTED** [review]
I think this looks mostly fine!

File: src/GitVersion.Core.Tests/VersionCalculation/Strategies/MergeMessageBaseVersionStrategyTests.cs (hunk 2 above)
Comment (DSpirit) [reply src/GitVersion.Core.Tests/VersionCalculation/Strategies/MergeMessageBaseVersionStrategyTests.cs:69]: Yes, the `is-release-branch` is set to `true`.
Here's an excerpt of the generated configuration:

``` yaml
hotfix:
      mode: ManualDeployment
      label: beta
      increment: Inherit
      prevent-increment:
        when-current-commit-tagged: false
      regex: ^hotfix(es)?[/-](?<BranchName>.+)
      source-branches:
      - release
      - main
      - support
      - hotfix
      is-source-branch-for: []
      is-release-branch: true
      pre-release-weight: 30000
```

Unfortunately I can't tell whether this is on purpose or not.

File: src/GitVersion.Core/MergeMessage.cs (hunk 1 above)
Comment (DSpirit) [reply src/GitVersion.Core/MergeMessage.cs:99]: Understood, I've changed it.

**Comment by DSpirit** [comment]
Closing this PR in favour to https://github.com/GitTools/GitVersion/pull/3984 with all current changes from master.

**Pull request closed without merging**